### PR TITLE
Add get_command_list methods to command builders

### PR DESCRIPTION
**Pull request by VakarisZ**

### CODE REVIEW

**Review by ilija-lazoroski — APPROVED** [review]

File: agentpluginapi/i_linux_agent_command_builder.py [hunk 1]
```diff
@@ -58,11 +58,17 @@ def build_run_command(self, run_options: LinuxRunOptions):
...
+        """
+
+    @abc.abstractmethod
+    def get_command_list(self) -> List[str]:
+        """
+        Returns the list of built commands
         """
 
     @abc.abstractclassmethod
-    def reset_command(self):
```
Comment (ilija-lazoroski) [inline agentpluginapi/i_linux_agent_command_builder.py:65]: I think this was intentional. Just `reset` is not really clear what it is resetting.

File: agentpluginapi/i_linux_agent_command_builder.py [hunk 2]
```diff
@@ -58,11 +58,17 @@ def build_run_command(self, run_options: LinuxRunOptions):
     @abc.abstractmethod
     def get_command(self) -> str:
         """
-        Gets the resulting command
+        Joins built commands into a one-liner command and returns it
+        """
+
+    @abc.abstractmethod
+    def get_command_list(self) -> List[str]:
```
Comment (ilija-lazoroski) [inline agentpluginapi/i_linux_agent_command_builder.py:65]: Use just `list`

**Review by mssalvatore — CHANGES_REQUESTED** [review]
I'm not sure I understand the need for this change. The command builder is designed to do:

```python
command_builder.build_download_command(my_options)
download_command = command_builder.get_command()
command_builder.reset()

command_builder.build_run_command(my_options)
run_command = command_builder.get_command()
command_builder.reset()
```

File: agentpluginapi/i_linux_agent_command_builder.py (hunk 1 above)
Comment (mssalvatore) [reply agentpluginapi/i_linux_agent_command_builder.py:65]: Additionally it's consistent. All the methods are "build_this_command", "build_that_command", "get_command". So "reset_command" follows the same scheme.

**Comment by VakarisZ** [comment]
Hmmm... Perhaps it could be done like this. It seems to be less flexible and more verbose though.

**Pull request closed without merging**